### ISSUE TITLE
There is a problem with the CapsLk key switching between Chinese and English at Big Sur beat5 and Big Sur beat6

### ISSUE THREAD

**Issue by dmlzj** [acidanthera/bugtracker#1136]
- OpenCore 0.6.0
- VoodooPS2 2.1.6
- VoodooInput 1.0.7

Hello, after upgrading my laptop (y7000) to big sur beat 5, there is a problem with the CapsLk key to switch between Chinese and English. It is normal on version 10.15.6!
The specific questions are:

![99999](https://user-images.githubusercontent.com/3021601/92321200-b5709a80-f05a-11ea-819c-8606135ce3b3.png)

- When this is not checked, you can directly press the CapsLK key to switch between Chinese and lowercase letters; for uppercase, holding down the shift key has an effect; long pressing the CapsLK key is invalid.
- After checking, pressing the CaplLK key becomes a switch between Chinese and uppercase letters. Long press is invalid. Only Ctrl + space can be switched to lowercase.
In version 10.15.6, switching between Chinese and English input methods is completely fine, and the experience is the same as that of White Apple!
- FN + Esc can switch whether to lock Fn, but the indicator light has not changed, which is normal in version 10.15.6!

**Comment by Mateo1234454545** [comment]
I do have this problem ,too.

**Comment by vit9696** [comment]
Beta problems.

**Comment by dmlzj** [comment]
@vit9696 
Hello, this is no problem on the White Apple system!

**Comment by dmlzj** [comment]
请问这个问题改好了吗？我还在用着10.15呢，后来说是笔记本是usb键盘的问题。。。不晓得有没有解决方案？

**Comment by mikebeaton** [comment]
> 请问这个问题改好了吗？我还在用着10.15呢，后来说是笔记本是usb键盘的问题。。。不晓得有没有解决方案？

Please provide your own translation, working language of the project is English.

> "Has this question been fixed? I'm still using 10.15, and later it was said that the laptop was the problem with the usb keyboard. . . Do not know if there is a solution?"

This issue has only been reported here as happening in an early beta which has long since been superseded. Please try in a non-beta version of Bug Sur. Report back if there is any issue. If you do not wish to upgrade without knowing for sure, perhaps someone who knows will comment here, otherwise please go to forums to ask for support/info about this.

**Comment by dmlzj** [comment]
I have installed the system of version 12.2 and tried it. It is said that the built-in keyboard of this notebook is USB interface, so it does not need the driver of PS2. I hope this problem can be solved. If necessary, I can provide my current EFI. @mikebeaton

**Comment by mikebeaton** [comment]
The problem still occurs on 12.2? Yes, please provide EFI.

Are you really certain that this is incorrect behaviour for this version of macOS in Chinese, not just a change from previous version?

**Comment by dmlzj** [comment]
Yes 12.2, I downloaded the full installation package to install it.
[EFI.zip](https://github.com/acidanthera/bugtracker/files/8131421/EFI.zip)

**Comment by mikebeaton** [comment]
Possibilities (not mutually exclusive):

 1. There is problem with OpenCore (&/or some loaded keyboard driver)
 1. There is a problem with your keyboard (&/or firmware)
 1. How macOS behaves has changed (in Big Sur upwards), and this behaviour is now normal

Please can you clarify which systems you are aware of where this behaviour does _not_ happen, particularly to rule out 3.

**Comment by dmlzj** [comment]
At present, the system version I am using is 10.15.7，there is no such problem. I think the system version is 10.15* ,there will be no such problem.
I have newly installed version 12.2 in other partitions of the hard disk,this problem is obvious!

&nbsp;In addition, this problem only exists in the built-in keyboard of the notebook, and the external keyboard is normal.
&nbsp;My laptop model is Lenovo y7000 10750h. It is said that the built-in keyboard interface of this model is USB, so there are no PS2 related files in my driver, only the driver related to the touch pad!



---Original---
From: "Mike ***@***.***&gt;
Date: Thu, Feb 24, 2022 21:55 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [acidanthera/bugtracker] There is a problem with the CapsLk key switching between Chinese and English at Big Sur beat5 and Big Sur beat6 (#1136)




 
Possibilities (not mutually exclusive):
  
There is problem with OpenCore (&amp;/or some loaded keyboard driver) 

There is a problem with your keyboard (&amp;/or firmware) 

How macOS behaves has changed (in Big Sur upwards), and this behaviour is now normal 
 
Please can you clarify which systems you are aware of where this behaviour does not happen, particularly to rule out 3.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by mikebeaton** [comment]
I've just done some own research - the option you talk about is not on English language macOS by default, but appears if I add Chinese keyboard input:
<img width="780" alt="Screenshot 2022-02-25 at 11 23 46" src="https://user-images.githubusercontent.com/11946605/155707197-df31119b-bdc8-483b-87ac-9f97165053b7.png">

After I have ticked it, I do not really know if it is behaving correctly, but:

 - If I press CAPS LOCK, the keyboad input symbol in the menu bar changes to Chinese
 - If I press CAPS LOCK again, the symbol changes back to English

So I guess it is working? This works on my MBP with Bug Sur 11.6.5, it also works on my hack with Monterey 12.2 with a USB keyboard.

So if you have problems with this feature, I wonder whether this is a change in macOS (since Bug Sur beta 5) which is responding to your laptop keyboard differently.

I have found a potentially useful utility to investigate further, from [this SO answer](https://apple.stackexchange.com/a/419678/113758) - which recommends [Karabiner Elements](https://karabiner-elements.pqrs.org/) keyboard viewer. I have no connection to that project and no guarantee that it is safe, but I believe it is. The included keyboard event view (just install s/ware, then start the viewer) does indeed seem to be very helpful:

<img width="1099" alt="Screenshot 2022-02-25 at 11 22 22" src="https://user-images.githubusercontent.com/11946605/155708362-dab037ec-5c26-4bce-8130-590e9152a773.png">

Note in particular, key codes are shown when CAPS LOCK is pressed and released.

My guess would be that your internal keyboard is NOT sending these keycodes, as compared to your external keyboard which is. You could use this viewer program to test this.

**Comment by dmlzj** [comment]
> So I guess it is working? This works on my MBP with Bug Sur 11.6.5, it also works on my hack with Monterey 12.2 with a USB keyboard.

Your MBP should not have a problem, ginkgo does not have this problem;
My black Apple laptop doesn't have this problem with an external USB keyboard.
The specific questions are:
CapsLk does switch from Chinese to English, but the uppercase English should be lowercase, long press will not respond to the uppercase English, if you want to switch to lowercase English, you need to press Ctrl+Shift key to switch;This description doesn't fully explain its problems. The actual situation is even worse, and the language switching is a bit messy. . .

The normal situation would be:
Press CapsLk to switch to English lowercase, press again to switch back to Chinese, long press once to switch to case.

Let me try to record a video.

**Comment by dmlzj** [comment]
---Original---
From: "Mike ***@***.***&gt;
Date: Fri, Feb 25, 2022 19:38 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [acidanthera/bugtracker] There is a problem with the CapsLk key switching between Chinese and English at Big Sur beat5 and Big Sur beat6 (#1136)




 
I've just done some own research - the option you talk about is not on English language macOS by default, but appears if I add Chinese keyboard input:
 
 
After I have ticked it, I do not really know if it is behaving correctly, but:
  
If I press CAPS LOCK, the keyboad input symbol in the menu bar changes to Chinese 

If I press CAPS LOCK again, the symbol changes back to English 
 
So I guess it is working? This works on my MBP with Bug Sur 11.6.5, it also works on my hack with Monterey 12.2 with a USB keyboard.
 
So if you have problems with this feature, I wonder whether this is a change in macOS (since Bug Sur beta 5) which is responding to your laptop keyboard differently.
 
I have found a potentially useful utility to investigate further, from this SO answer - which recommends Karabiner Elements keyboard viewer. I have no connection to that project and no guarantee that it is safe, but I believe it is. The included keyboard event view (just install s/ware, then start the viewer) does indeed seem to be very helpful:
 

 
Note in particular, key codes are shown when CAPS LOCK is pressed and released.
 
My guess would be that your internal keyboard is NOT sending these keycodes, as compared to your external keyboard which is. You could use this viewer program to test this.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
 
  	
 	 		  			从QQ邮箱发来的超大附件 		 	
 	 		  	 	 		 			 		
 		 			 VID_20220225_211434.mp4 (12.57M, 无限期) 			 			 				进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=cf9b0a35cf0a9f73fde84c3537386563874fab3535386563154b52060209535105074f02070c564e045d01001859500654495054540f520153070006540a7263662d266a07085751005650006a0a54520457561b584851772ead2cd96a17fc8cab608550e26f0de9972f8842&t=exs_ftn_download&code=0db558ec 			
 		
 	
 
  	 	 		 			 		
 		 			 VID_20220225_211715.mp4 (22.52M, 无限期) 			 			 				进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=cccd09309957cf75febe4f3061653565446a093163653565161d025305520303040b4c0755575748075002534e5c0403561f0303065206075504575405562265657b256f51550757030053053c5704540403541e0e1501712fb678bac712e7245950b1088354e0c8f3ed8d9b&t=exs_ftn_download&code=32a0ce5e

**Comment by dmlzj** [comment]
I have recorded a video of the two systems, which can be compared. 
The most obvious thing is that it becomes uppercase when you press CapsLK to switch English on the 12.2 system. 
You can pay attention to the switching state of the input method in the upper right corner, which is not normal under the 12.2 system.

**Comment by mikebeaton** [comment]
Okay, I think I understand the difference. I still works on both my systems (long press to change case, quick press to switch languages). Ofc my Dell hack is USB kb only, so not comparable. My MBP requires OpenCore to run Big Sur so may (or may not) be more comparable to your laptop, but it also works.

I suggested again using the Karabiner-EventViewer.app which is installed for free when installing Karabiner Elements - link above. Then you can examine whether your two keyboards produced the same scan codes or different, when CAPS LOCK is held, and when CAPS LOCK is pressed.

**Comment by dmlzj** [comment]
---Original---
From: "Mike ***@***.***&gt;
Date: Fri, Feb 25, 2022 21:42 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [acidanthera/bugtracker] There is a problem with the CapsLk key switching between Chinese and English at Big Sur beat5 and Big Sur beat6 (#1136)




 
Okay, I think I understand the difference. I still works on both my systems (long press to change case, quick press to switch languages). Ofc my Dell hack is USB kb only, so not comparable. My MBP requires OpenCore to run Big Sur so may (or may not) be more comparable to your laptop, but it also works.
 
I suggested again using the Karabiner-EventViewer.app which is installed for free when installing Karabiner Elements - link above. Then you can examine whether your two keyboards produced the same scan codes or different, when CAPS LOCK is held, and when CAPS LOCK is pressed.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dmlzj** [comment]
This keyboard code looks the same as yours. There should be no problem!




---Original---
From: "Mike ***@***.***&gt;
Date: Fri, Feb 25, 2022 21:42 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [acidanthera/bugtracker] There is a problem with the CapsLk key switching between Chinese and English at Big Sur beat5 and Big Sur beat6 (#1136)




 
Okay, I think I understand the difference. I still works on both my systems (long press to change case, quick press to switch languages). Ofc my Dell hack is USB kb only, so not comparable. My MBP requires OpenCore to run Big Sur so may (or may not) be more comparable to your laptop, but it also works.
 
I suggested again using the Karabiner-EventViewer.app which is installed for free when installing Karabiner Elements - link above. Then you can examine whether your two keyboards produced the same scan codes or different, when CAPS LOCK is held, and when CAPS LOCK is pressed.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by mikebeaton** [comment]
Not only the same scancode _value_, but scancodes arriving with the same timing (when pressing and releasing the CL key) - that is what I am guessing needs to to be the same, to get the same behaviour, so I am guessing it is different between your two keyboards.

But you are saying there is no difference in scan code presence or timing between your two keyboards? Surprising...

To me this is looking like an issue with the particular hardware, I cannot see any clear evidence at all of a bug in OpenCore or associated tools.

In order to get to any possible fix, you will need to find some underlying difference in keyboard behaviour which macOS is responding to, and then work out how to fix or alter it.

Maybe there are some additional BIOS settings for your laptop's internal keyboard?

**Comment by dmlzj** [comment]
Thanks anyway for helping me with this problem.
I don't know how to find out the difference between them, I just connected the external keyboard, when the external keyboard was switched to Chinese, I pressed the internal keyboard, but it was still in English input state, this is a very strange problem.
Probably because of any changes in keyboard behavior that have been made since system 11. I really hope we can solve this problem.
